src/components/Board.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useContext } from 'react'
+import Square from './Square'
+import Confetti from 'react-dom-confetti'
+import confettiConfig from './confetti-config'
+import { GameStateContext } from './Game'
+import { checkWinner, checkIsTied, checkIsHighlightedSquare } from 'utils'
+
+//component, destructured props of type BoardTypeProps
+const Board = () => {
+  const { gameState, stateDispatch } = useContext(GameStateContext)
+
+  useEffect(() => {
+    const isWinningSquareSet = checkWinner(gameState.currentBoardState)
+
+    if (isWinningSquareSet) {
+      //one player won
+      stateDispatch({
+        type: 'PLAYER_WON',
+        newGameState: { ...gameState, winningSquareSet: isWinningSquareSet },
+      })
+    } else if (checkIsTied(gameState.currentBoardState) && !gameState.isWon) {
+      //no one won, tied
+      stateDispatch({ type: 'TIE', newGameState: gameState })
+    } else {
+      //in progress game
+      stateDispatch({
```

Comment:
shouldn't need this, incrementing isXNext should be closer to your resolving turn event handlers